### ISSUE TITLE
headscale: enable magicdns

### ISSUE THREAD

**Issue by majst01** [metal-stack/metal-roles#108]
this will make things easier because we must not figure out ips anymore, instead we can just connect to the machineID

**Comment by GrigoriyMikhalkin** [comment]
I remember i tried to use MagicDNS. The main problem was adapting `metalctl firewall ssh` command: https://github.com/metal-stack/metalctl/pull/148#issuecomment-1218522714

**Comment by majst01** [comment]
Yes but i would still enable it in the headscale server to have at least the option.
